nil pointer dereference

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0xa0 pc=0x559982]

goroutine 1 [running]:
panic(0xaeae80, 0xc820010060)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
project/vendor/github.com/spf13/viper.(*Viper).Sub(0xc8200ee380, 0xc01f18, 0x3, 0xc82003df10)
	/project/vendor/github.com/spf13/viper/viper.go:528 +0x372
```

```go
func (v *Viper) Sub(key string) *Viper {
	subv := New()
	data := v.Get(key)
	if reflect.TypeOf(data).Kind() == reflect.Map {  <----- reflect.TypeOf(data) may return nil, especially if key is not set
		subv.config = cast.ToStringMap(data)
		return subv
	} else {
		return nil
	}
}
```

--- a/viper.go
+++ b/viper.go
@@ -532,6 +532,10 @@ func Sub(key string) *Viper { return v.Sub(key) }
 func (v *Viper) Sub(key string) *Viper {
 	subv := New()
 	data := v.Get(key)
+	if data == nil {
+		return nil
+	}
+
 	if reflect.TypeOf(data).Kind() == reflect.Map {
 		subv.config = cast.ToStringMap(data)
 		return subv

--- a/viper_test.go
+++ b/viper_test.go
@@ -760,6 +760,9 @@ func TestSub(t *testing.T) {
 
 	subv = v.Sub("clothing.pants.size")
 	assert.Equal(t, subv, (*Viper)(nil))
+
+	subv = v.Sub("missing.key")
+	assert.Equal(t, subv, (*Viper)(nil))
 }
 
 var yamlMergeExampleTgt = []byte(`

